Include/internal/pycore_code.h

```diff
@@ -205,6 +114,13 @@ _Py_IncrementCountAndMaybeQuicken(PyCodeObject *code)
 
 extern Py_ssize_t _Py_QuickenedCount;
 
+// Borrowed references to common callables:
+struct callable_cache {
+    PyObject *isinstance;
+    PyObject *len;
+    PyObject *list_append;
+};
```

Comment:
I think the existence of `PyList_Type` as part of the API means that `list.append` must be per-process unique.
In other words, `list_append` could be static.

I'm happy to leave it as is for now, though. We should look to make the whole struct static, although the mutability of builtin functions makes that tricky for `isinstance` and `len`.

Reply:
I believe each interpreter has its own `builtins` module (check out `_PyBuiltin_Init`), so making this static could be tricky. As you said, though: probably worth looking into in the future.